x-pack/plugins/security_solution/public/risk_score/containers/deprecated/index.ts

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useCallback, useEffect, useMemo } from 'react';
+import { REQUEST_NAMES, useFetch } from '../../../common/hooks/use_fetch';
+import { RiskQueries } from '../../../../common/search_strategy';
+import type { Params, Response } from './api';
+import { getRiskScoreDeprecated, RiskEntity } from './api';
+
+interface RiskScoresDeprecated {
+  error: unknown;
+  isDeprecated: boolean;
+  isEnabled: boolean;
+  isLoading: boolean;
+  refetch: (indexName: string) => void;
+}
+
+export const useRiskScoreDeprecated = (
+  isFeatureEnabled: boolean,
+  factoryQueryType: RiskQueries.hostsRiskScore | RiskQueries.usersRiskScore,
+  defaultIndex?: string
+): RiskScoresDeprecated => {
+  const entity = useMemo(
+    () => (factoryQueryType === RiskQueries.hostsRiskScore ? RiskEntity.host : RiskEntity.user),
+    [factoryQueryType]
+  );
+
+  const { fetch, data, isLoading, error } = useFetch<Params, Response, unknown>(
```

Comment:
NIT: No need to pass the types explicitly, TS will take them from the `getRiskScoreDeprecated` function params and response types
```suggestion
  const { fetch, data, isLoading, error } = useFetch(
```

Reply:
I'll have to apply useFetch to my PR.